### PR TITLE
Move inheritedOfType from Component to Widget so that Switch can use it

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

File: sky/packages/sky/lib/widgets/framework.dart [hunk 1]
```diff
@@ -272,6 +272,22 @@ abstract class Widget {
...
+    if (_dependencies == null)
+      _dependencies = new Set<Type>();
+    _dependencies.add(targetType);
+    Widget ancestor = parent;
+    while (ancestor != null && ancestor.runtimeType != targetType)
+      ancestor = ancestor.parent;
+    return ancestor;
+  }
+
+  void dependenciesChanged() {
```
Comment (eseidelGoogle) [inline sky/packages/sky/lib/widgets/framework.dart]: Can this be abstract?

File: sky/packages/sky/lib/widgets/switch.dart [hunk 2]
```diff
@@ -30,46 +30,28 @@ const Duration _kCheckDuration = const Duration(milliseconds: 200);
...
       : super(key: key);
 
   final bool value;
   final ValueChanged onChanged;
-  final Color thumbColor;
 
   _RenderSwitch get renderObject => super.renderObject;
   _RenderSwitch createNode() => new _RenderSwitch(
-      value: value, thumbColor: thumbColor, onChanged: onChanged);
+      value: value, thumbColor: null, onChanged: onChanged);
```
Comment (Hixie) [inline sky/packages/sky/lib/widgets/switch.dart]: wrap this by indenting two from the previous line, one argument to a line, final ) on its own line

File: sky/packages/sky/lib/widgets/switch.dart [hunk 3]
```diff
@@ -30,46 +30,28 @@ const Duration _kCheckDuration = const Duration(milliseconds: 200);
...
     renderObject.onChanged = onChanged;
-    renderObject.thumbColor = thumbColor;
+    renderObject.thumbColor = Theme.of(this).accentColor;
   }
 }
 
 class _RenderSwitch extends RenderToggleable {
   _RenderSwitch(
-      {bool value, Color thumbColor: _kThumbOffColor, ValueChanged onChanged})
+      { bool value, Color thumbColor: _kThumbOffColor, ValueChanged onChanged })
```
Comment (Hixie) [inline sky/packages/sky/lib/widgets/switch.dart]: { on the previous line, each argument on its own line indented two from the first line, final }) on its own line

**Comment by Hixie** [comment]
LGTM.

Is there some way you can compare runtime memory usage before and after this change, so we can get a sense of the cost?

**Comment by collinjackson** [comment]
I tried running "top" and there wasn't a huge difference. We should probably have better memory profiling tools.

File: sky/packages/sky/lib/widgets/framework.dart (hunk 1 above)
Comment (collinjackson) [inline sky/packages/sky/lib/widgets/framework.dart]: No, but I updated the comment to make more sense

**Pull request merged**